decimal.go

```diff
@@ -32,322 +23,78 @@ const (
 	RoundBankers                   // Round to nearest, on tie round to even number
 )
 
-var logTable = []int64{
-	1,
-	10,
-	100,
-	1000,
-	10000,
-	100000,
-	1000000,
-	10000000,
-	100000000,
-	1000000000,
-	10000000000,
-	100000000000,
-	1000000000000,
-	10000000000000,
-	100000000000000,
-	1000000000000000,
-	10000000000000000,
-	100000000000000000,
-	1000000000000000000,
-}
-
-// Zero create a new decimal number that is equal to zero.
-func Zero() Number {
-	return Number{}
-}
-
-// New creates a new decimal number having value of val*10^exp.
-func New(val int64, exp int) Number {
-	return Number{val, exp}
-}
-
-// FromInt creates a new instance of decimal number with an integer value and
-// zero exponent.
-func FromInt(val int) Number {
-	return Number{int64(val), 0}
-}
-
-// FromString creates a new instance of decimal number by parsing given string.
-//
-// Accepted format examples: "1234", "123.456". "123.000".
-//
-// Scientific notation is not supported.
-func FromString(str string) (Number, error) {
-	d := Number{}
-	err := d.UnmarshalText([]byte(str))
-	return d, err
+func init() {
+	newDecimal.MarshalJSONWithoutQuotes = true
 }
 
-// FromRat creates a new instance of decimal number from a Rat value with
-// given exponent.
-func FromRat(val *big.Rat, exp int) Number {
-	f, _ := val.Float64()
-	f /= math.Pow10(exp)
-	digits := int(math.Log10(f))
-	if digits > 18 {
-		exp += (digits - 18)
-		f /= math.Pow10(digits - 18)
+// NewFromRat this only works for small Rats, big.Rats won't necessarily fit into a Float64,
+// so you'll be losing information here.
+// https://github.com/shopspring/decimal/pull/4/files
+func NewFromRat(value *big.Rat, exp int) newDecimal.Decimal {
```

Comment:
Perhaps, it would be simpler to omit `exp` attribute and use some sort of precision variable instead? 
https://github.com/shopspring/decimal/pull/288/files

We could also use a predefined precision here, so the second parameter could be omitted altogether.
https://github.com/shopspring/decimal/blob/master/decimal.go#L45 precision.